functionary/ui/views/view_base.py

```diff
@@ -0,0 +1,49 @@
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.views.generic.detail import DetailView
+from django.views.generic.list import ListView
+
+from core.auth import Permission
+from core.models import Environment
+
+
+class PermissionedEnvironmentListView(ListView):
+    model_field = "environment"
+    environment_through_field = None
+    order_by_fields = ["name"]
+
+    def get_queryset(self):
+        """Filters out object not in the environment then sorts based
+        on the value of order_by_fields."""
+
+        env_id = self.request.session["environment_id"]
+        env = Environment.objects.get(id=env_id) if len(env_id) > 2 else None
+        if env:
+            if self.request.user.is_superuser or self.request.user.has_perm(
```

Comment:
The `has_perm` check would be `True` in the superuser case so  you don't need to check that explicitly.

Reply:
done